src/estimation/src/AttitudeEstimatorUtils.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) 2019 Fondazione Istituto Italiano di Tecnologia
+ *
+ * Licensed under either the GNU Lesser General Public License v3.0 :
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ * or the GNU Lesser General Public License v2.1 :
+ * https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * at your option.
+ */
+
+#include <iDynTree/Estimation/AttitudeEstimatorUtils.h>
+
+iDynTree::Vector3 crossVector(const iDynTree::Vector3& a, const iDynTree::Vector3& b)
+{
+    iDynTree::Vector3 out;
+    iDynTree::toEigen(out) = iDynTree::toEigen(a).cross(iDynTree::toEigen(b));
+    return out;
+}
+
+iDynTree::Matrix3x3 mapR3Toso3(const iDynTree::Vector3& omega)
+{
+    iDynTree::Matrix3x3 out;
+    iDynTree::toEigen(out) = iDynTree::skew(iDynTree::toEigen(omega));
+    return out;
+}
+
+bool checkSkewSymmetricity(const iDynTree::Matrix3x3& S)
+{
+    bool flag = false;
+    if ( (iDynTree::toEigen(S) + iDynTree::toEigen(S).transpose()).isZero() )
+    {
+        flag = true;
+    }
+    return flag;
+}
+
+iDynTree::Vector3 mapso3ToR3(const iDynTree::Matrix3x3& S)
+{
+    iDynTree::Vector3 out;
+    iDynTree::toEigen(out) = iDynTree::unskew(iDynTree::toEigen(S));
+    return out;
+}
+
+double innerProduct(const iDynTree::Vector3 a, const iDynTree::Vector3& b)
+{
+    return iDynTree::toEigen(a).dot(iDynTree::toEigen(b));
+}
+
+double realPartOfQuaternion(const iDynTree::Quaternion& q)
+{
+    return q(0);
+}
+
+iDynTree::Vector3 imaginaryPartOfQuaternion(const iDynTree::Quaternion &q)
+{
+    iDynTree::Vector3 v;
+    v(0) = q(1);
+    v(1) = q(2);
+    v(2) = q(3);
+    return v;
+}
+
+iDynTree::Quaternion composeQuaternion(const iDynTree::Quaternion &q1, const iDynTree::Quaternion &q2)
+{
+    // const clock_t begin_time = clock();
+    iDynTree::Quaternion out;
+    double s1 = realPartOfQuaternion(q1);
+    iDynTree::Vector3 v1 = imaginaryPartOfQuaternion(q1);
+
+    double s2 = realPartOfQuaternion(q2);
+    iDynTree::Vector3 v2 = imaginaryPartOfQuaternion(q2);
+
+    iDynTree::Vector3 imagOut;
+    iDynTree::toEigen(imagOut) = iDynTree::toEigen(v2)*s1 + iDynTree::toEigen(v1)*s2 + iDynTree::toEigen(crossVector(v1, v2));
+    out(0) = s1*s2 - innerProduct(v1, v2);
+    out(1) = imagOut(0);
+    out(2) = imagOut(1);
+    out(3) = imagOut(2);
+    //std::cout << "average time : " << float( clock() - begin_time ) / CLOCKS_PER_SEC << " s." << std::endl;
+    return out;
+}
+
+iDynTree::Matrix4x4 mapofYQuaternionToXYQuaternion(const iDynTree::Quaternion &x)
+{
+    // unitary matrix structure represented by 2 complex numbers z1 = q0+iq1 and z2 = q2+iq3
+    std::vector<double> v{x(0), -x(1), -x(2), -x(3),
```

Comment:
As indicated by the AppVeyor failure, you are using `std::vector` here without ever including the `<vector>` standard C++ header. In *nix, compilation is just working because of some lucky transitive include.

Reply:
I think the Windows errors are only due to the missing header definitions
``` cpp
#define _USE_MATH_DEFINES
#define <cmath>
#include <vector>
```
Would you agree?

Reply:
Yes, let's try with that and see if it fixes the CI.